### PR TITLE
Fix parsing and bring package up to date with 2022 reality

### PR DESCRIPTION
**Pull request by filipsch**
See the separate commits for some info

### CODE REVIEW

File: R/processing.R [hunk 1]
```diff
@@ -57,13 +57,12 @@ parse_description <- function(pkg_folder, pkg_url, repo_type) {
...
-  file.rename(file.path(pkg_folder, "vignettes"), file.path(pkg_folder, "_vignettes"))
+  # file.rename(file.path(pkg_folder, "vignettes"), file.path(pkg_folder, "_vignettes"))
   pkg <- pkgdown:::as_pkgdown(pkg_folder)
-  pkg$topics %>%
-    transpose() %>%
-    map(pkgdown:::data_reference_topic, pkg, examples = FALSE) %>%
-    map(clean_up) %>%
-    map(add_pkg_info, pkg_folder)
+  x <- transpose(pkg$topics)
+  y <- map(x, pkgdown:::data_reference_topic, pkg)
```
Comment (ztsorojev) [inline R/processing.R]: Should we also have `examples = FALSE` inside that map?
Not sure what its doing

File: R/processing.R [hunk 2]
```diff
@@ -57,13 +57,12 @@ parse_description <- function(pkg_folder, pkg_url, repo_type) {
 #' @export
 parse_topics <- function(pkg_folder) {
   message("Parsing topics ...")
-  file.rename(file.path(pkg_folder, "vignettes"), file.path(pkg_folder, "_vignettes"))
+  # file.rename(file.path(pkg_folder, "vignettes"), file.path(pkg_folder, "_vignettes"))
```
Comment (ztsorojev) [inline R/processing.R]: Do you know why this rename was here?

File: R/utils.R [hunk 3]
```diff
@@ -6,17 +6,19 @@ delete_files <- function() {
...
-  untar(tar_path, exdir = "packages/")
-  file.remove(tar_path)
+  untar_dir <- file.path(tmp_dir, pkg_name)
+  untar(tar_path, exdir = untar_dir)
+  # file.remove(tar_path)
 
-  # possible that tar was not unpacked with package name
-  pkg_folder <- file.path("packages", pkg_name)
-  file.rename(list.dirs("packages", recursive = FALSE), pkg_folder)
+  pkg_folder <- list.dirs(untar_dir, recursive = FALSE)
```
Comment (ztsorojev) [inline R/utils.R]: Does this change some of the functionality or it's just a refactoring?
It seems to be the same for me from what I understand but I am not familiar with R

**Review by ztsorojev — COMMENTED** [review]
Just left some small comments, I know the PR is not finalized yet. Let's talk tomorrow! :)

File: R/utils.R (hunk 3 above)
Comment (filipsch) [reply R/utils.R]: There's some refactoring here so we're using temporary folders.

**Review by ztsorojev — APPROVED** [review]
@filipsch It looks good, I tested with SQS queues and it works! 🙌  I can post messages to the rdoc-r-worker queue, poll them locally, run the main() function and it successfully adds the messages to the rdoc-app-worker queue read by the API!

Only thing I noticed is that the test for rbase is commented, when I tried to run it, it fails. I think you said that one is less critical since it doesn't change that often but I can't remember for sure. If it's fine, then we can deploy! The circleci jobs are passing, we just need to merge to master and it will deploy to prod!

**Comment by filipsch** [comment]
@ztsorojev I did a PR to update the `parser_version` to `2` now that we have made pretty big updates.

**Pull request merged**